joint_state_controller/include/joint_state_controller/joint_state_controller.hpp

```diff
@@ -27,11 +29,16 @@
 
 #include "sensor_msgs/msg/joint_state.hpp"
 
+class JointStateControllerTest;
+
 namespace joint_state_controller
 {
 
 class JointStateController : public controller_interface::ControllerInterface
 {
+private:
+  friend class ::JointStateControllerTest;
```

Comment:
could we please follow the inheritance-based setup for this? would really like to avoid friends

Reply:
We can do inheritance. However, since I'm testing member variables this will require switching to `protected` instead of `private` in the `JointStateController` class. Is this acceptable?

Reply:
We can. But then we need the member variables to be `protected` instead of `private` in order to test them (and we do a lot of that in these tests!). Acceptable?

Reply:
absolutely

Reply:
Done.

Reply:
thank you!